### PR TITLE
spirv: emit vectorized operations

### PR DESCRIPTION
**Pull request by alichraghi**

### CODE REVIEW

File: src/codegen/spirv.zig [hunk 1]
```diff
@@ -1774,29 +1817,28 @@ const DeclGen = struct {
...
+        /// vectors, fake vectors (arrays) and single elements, depending on `result_ty`.
         /// After calling this function, this WIP is no longer usable.
         /// Results is in `direct` representation.
         fn finalize(wip: *WipElementWise) !IdRef {
-            if (wip.is_vector) {
+            if (wip.is_array) {
                 // Convert all the constituents to indirect, as required for the array.
                 for (wip.results) |*result| {
-                    result.* = try wip.dg.convertToIndirect(wip.scalar_ty, result.*);
+                    result.* = try wip.dg.convertToIndirect(wip.ty, result.*);
```
Comment (Snektron) [inline src/codegen/spirv.zig:1841]: The results of finalize() should be direct representation, so I think this should just call constructComposite directly.

Comment (Snektron) [reply src/codegen/spirv.zig:1841]: Or did you not yet implement that part?

Comment (alichraghi) [reply src/codegen/spirv.zig:1841]: I didn't, I was afraid that it will get big and complicated like the last two attempts

Comment (Snektron) [reply src/codegen/spirv.zig:1841]: ok, we'll do it in a different pr

**Review by Snektron — APPROVED** [review]

**Pull request merged**